### PR TITLE
Dev: Release Script Fix

### PR DESCRIPTION
**Pull request by RussellSpitzer**
Default is wrong on remote
Key values are not used

### CODE REVIEW

File: dev/source-release.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ usage () {
 }
 
 # Default repository remote name
-remote="origin"
+remote="apache"
```
Comment (RussellSpitzer) [inline dev/source-release.sh:39]: https://github.com/apache/iceberg/pull/4919/files#diff-ee187088d9f56516cc2abd9e4b4294a92d23dc09746db5b0976799131c293d2bR33

Comment (kbendick) [reply dev/source-release.sh:39]: I think this depends on where the script gets run (in the case of origin) and then how the user / box has their git remote set up.

Like mine is `upstream` vs apache, but I know most people / PMC members specifically probably use apache.

Comment (RussellSpitzer) [reply dev/source-release.sh:39]: Sorry, the link is meant to show this line above
```    echo "  -g      Specify Git remote name. Defaults to \"apache\""```

File: dev/source-release.sh [hunk 2]
```diff
@@ -112,7 +112,7 @@ tarball=$tag.tar.gz
 git archive $release_hash --worktree-attributes --prefix $tag/ -o $projectdir/$tarball
 
 echo "Signing the tarball..."
-[[ -z "$keyid" ]] && keyopt="-u $keyid"
+[[ -n "$keyid" ]] && keyopt="-u $keyid"
```
Comment (kbendick) [inline dev/source-release.sh:115]: 👍

File: dev/source-release.sh (hunk 1 above)
Comment (kbendick) [reply dev/source-release.sh:39]: Ahhh! Ok then absolutely yes let’s make it apache. This typically gets run manually in somebody’s machine, so `origin` is very likely incorrect (typically one’s fork).

But if it says the default is `apache`, it should be that.

**Comment by kbendick** [comment]
Also as an aside, during the _actual_ release process, we need to tag the SHA we build off of in master as `release-base-X.Y.0` for a new (non-patch) release.

I don’t think that step is automated anywhere but it’s how the next SNAPSHOT version gets its name (via a function in build.gradle).

File: dev/source-release.sh (hunk 2 above)
Comment (kbendick) [reply dev/source-release.sh:115]: What I've done in the past is add a function that logs either way, so logs X on true and Y on false, so we'd get `No keyid is found in variable \\$keyid, skipping adding signature to tarball` in the untrue path (just some indicator it was taken).

Comment (RussellSpitzer) [reply dev/source-release.sh:115]: Yeah trick here is missing that arg still signs, just with the default key

Comment (kbendick) [reply dev/source-release.sh:115]: Oh haha you’re right. This is why I advocate for logging somewhat verbosely in shell scripts.

**Pull request merged**